### DISCUSSION TITLE
Real-time cost or event-history

### DISCUSSION THREAD

**Discussion by wind7water** [amzn/ads-advanced-tools-docs#57] [Q&A]
hi here,

I have been working in advertisement API.I wander if there is any way to get real-time cost(or almost real-time)?

And I need the operations history of the Sponsored(like the picture below).How can I get them?

thanks
![28D05741-CFBF-4916-80A2-3D2598084BF3](https://user-images.githubusercontent.com/128784023/227450858-244d782d-160c-4d3e-88f5-58f38fe98da9.png)

**Comment by kdwivedi13** [comment]
Did you have the chance to check out Amazon Marketing Stream? If you're looking for something real-time or nearly real-time, Amazon Marketing Stream is the way to go. 

Please visit [developer portal](https://advertising.amazon.com/API/docs/en-us/amazon-marketing-stream/overview) to learn more about Amazon marketing stream and its supported datasets.

If you find it useful then do explore reference application, it will stream integration much easier for you.
https://github.com/amzn/amazon-marketing-stream-examples

I hope this information is useful for you.

**Reply by wind7water** [reply]
I use the Campaign reports(https://advertising.amazon.com/API/docs/en-us/reporting/v3/report-types) to get the cost of Sponsored Products.But the data I got was delayed,and even delayed than the data in webside.
So I wander if there is an api to get the real-time cost of my Sponsored Products.

And by the way,I need the operations history of the Sponsored.which api will works?
